metricflow-semantics/metricflow_semantics/model/semantics/linkable_element_set.py

```diff
@@ -402,8 +408,10 @@ def filter_by_spec_patterns(self, spec_patterns: Sequence[SpecPattern]) -> Linka
             if LinkableElementSet._path_key_to_spec(path_key) in specs_to_include:
                 path_key_to_linkable_metrics[path_key] = linkable_metrics
 
-        return LinkableElementSet(
+        filtered_elements = LinkableElementSet(
             path_key_to_linkable_dimensions=path_key_to_linkable_dimensions,
             path_key_to_linkable_entities=path_key_to_linkable_entities,
             path_key_to_linkable_metrics=path_key_to_linkable_metrics,
         )
+        logger.info(f"Filtering valid linkable elements took: {time.time() - start_time:.2f}s")
```

Comment:
I would love to cache this function too, but that change would be a bit more involved since base class (`LinkableElementSet`) relies heavily on dictionaries, which are not hashable.

Reply:
Can we just store `Sequence[Tuple[ElementPathKey, Sequence[LinkableDimension]]]` or whatever instead of a dict? We go to some lengths to avoid mutation of these things anyway, might as well make it official. 

We shouldn't do it in this PR, anyway, that sounds like a fairly involved change given the need to update callsites and all of those merge functions and whatnot.

Reply:
Yeah that could work!